notebooks/ir_scattered_light_calwf3_corrections/ima_visualization_and_differencing.py

```diff
@@ -0,0 +1,357 @@
+        Below each panel, we list the IMSET difference, along with the time interval between the two IMSETs.
+        The statistics in orange (on the left and right side of each panel) give the mean rate and 
+        standard deviation of each side of the image, respectively. The value in green 'delta' is the 
+        difference between the left and right side of the image. 
+        The value in white "Ratio" gives the ratio of the mean difference in orange 
+        for the left versus the right side. 
+    '''
+    
+    
+    xlabel_list = ["SCI[16-15]","SCI[15-14]","SCI[14-13]","SCI[13-12]","SCI[12-11]",
+                 "SCI[11-10]","SCI[10-9]","SCI[9-8]","SCI[8-7]","SCI[[7-6]]","SCI[6-5]",
+                "SCI[5-4]","SCI[4-3]","SCI[3-2]","SCI[2-1]"]
+    
+    fig, axarr = plt.subplots(4, 4)
+    fig.set_size_inches(40, 40)
+    fig.set_dpi(40)
+    itime = integ_time[0:-1] - integ_time[1:]
+
+    siglvl = (0.5,0.5)
+    
+    ima_j = cube[:, :, 1:]
+    inst_diff = ((ima_j*(integ_time[1:]-integ_time[0]))-\
+             (ima_j*(integ_time[0:-1]-integ_time[0])))/(integ_time[1:]-integ_time[0:-1])
+ 
+    
+    for i, ax in enumerate(axarr.reshape(-1)):
+        if (i < cube.shape[-1]-2):
+            i=i+1
+            text = ax.text(50, 500, f'{median_diff_lhs[i]:.3f}\n±\n{standard_dev_lhs[i]:.3f}', color='Orange', fontsize=30)
+            text.set_path_effects([path_effects.Stroke(linewidth=15, foreground='black'),
+                   path_effects.Normal()])
+            text = ax.text(700, 500, f'{median_diff_rhs[i]:.3f}\n±\n{standard_dev_rhs[i]:.3f}', color='Orange', fontsize=30)
+            text.set_path_effects([path_effects.Stroke(linewidth=15, foreground='black'),
+                   path_effects.Normal()])
+            text = ax.text(200, 900, f'Ratio = {median_diff_lhs[i]/median_diff_rhs[i]:.2f}', color='White', fontsize=30)
+            text.set_path_effects([path_effects.Stroke(linewidth=15, foreground='black'),
+                   path_effects.Normal()])
+            text = ax.text(300, 300, f'$\Delta = ${median_diff_lhs[i]-median_diff_rhs[i]:.2f}', color='#32CD32', fontsize=30)
+            text.set_path_effects([path_effects.Stroke(linewidth=15, foreground='black'),
+                   path_effects.Normal()])
+            
+            if diff_method == "cumulative":
+
+                diff = cube[:,:,i]-cube[:,:,i+1]
+                im = ax.imshow(np.abs(diff), cmap='Greys_r', origin='lower',
+                           vmin=median_diff_rhs[i]-siglvl[0]*median_diff_rhs[i], 
+                           vmax=median_diff_lhs[i]+siglvl[1]*median_diff_lhs[i])
+                ax.set_title(f'$\mu = ${np.nanmedian(cube[:,:,i]-cube[:,:,i+1]):.2f}±{standard_dev_fullframe[i]:.2f} e-/s', fontsize = 30)
+                
+            elif diff_method == "instantaneous":
+                diff = inst_diff
+                vmin,vmax = zscale(diff[:,:,i])
```

Comment:
its possible that getting the zscale for each diff is why the panels all look so similar.  Maybe just use the vmin and vmax for the read that shows the strongest TVB or something, and then keep the scale consistent for all the other reads.